### PR TITLE
feat(server): group domain models and GQL

### PR DESCRIPTION
**Pull request by mimoham24**

### CODE REVIEW

File: server/pkg/value/string.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ const TypeRichText Type = "richText"
 const TypeMarkdown Type = "markdown"
 const TypeSelect Type = "select"
 const TypeTag Type = "tag"
+const TypeGroup Type = "group"
```
Comment (rot1024) [inline server/pkg/value/string.go:17]: GroupID is not a string, it's id.GroupID. Refer to https://github.com/reearth/reearth-cms/blob/group/domain-models/server/pkg/value/reference.go#L12C4-L12C27

File: server/schemas/group.graphql [hunk 2]
```diff
@@ -0,0 +1,47 @@
+type Group implements Node {
+    id: ID!
+    schemaId: ID!
+    projectId: ID!
+    name: String!
+    description: String!
+    key: String!
```
Comment (yk-eukarya) [inline server/schemas/group.graphql:7]: is this key required? is not schema field key enough?

Comment (mimoham24) [reply server/schemas/group.graphql:7]: yes sure, If you check the Figma the group should have a key field in a similar way to the model
also, the group schema field is not a group.... the schema group field is pointing at the group but they are both different domains

File: server/schemas/field.graphql [hunk 3]
```diff
@@ -227,10 +240,12 @@ input SchemaFieldTypePropertyInput @onlyOne {
   integer: SchemaFieldIntegerInput
   reference: SchemaFieldReferenceInput
   url: SchemaFieldURLInput
+  group: SchemaFieldGroupInput
 }
 
 input CreateFieldInput {
-  modelId: ID!
+  modelId: ID
+  groupId: ID
```
Comment (yk-eukarya) [inline server/schemas/field.graphql:248]:
```suggestion
  schemaId: ID
```

File: server/schemas/group.graphql (hunk 2 above)
Comment (yk-eukarya) [reply server/schemas/group.graphql:7]: >  the group schema field is not a group.... the schema group field is pointing at the group but they are both different entities

this is clear to me, but how we will use the group key?

File: server/schemas/field.graphql (hunk 3 above)
Comment (mimoham24) [reply server/schemas/field.graphql:248]: I totally agree, but this change requires a lot of changes in our current implementation. Unfortunately, we have a critical deadline this time
I'm planning to refactor this part later

File: server/e2e/gql_field_test.go [hunk 4]
```diff
@@ -21,7 +21,7 @@ func createField(e *httpexpect.Expect, mID, title, desc, key string, multiple, u
 				  }
 				}`,
 		Variables: map[string]any{
-			"modelId":      mID,
+			"modelId":      &mID,
```
Comment (yk-eukarya) [inline server/e2e/gql_field_test.go:24]:
```suggestion
			"modelId":      mID,
```

File: server/e2e/gql_field_test.go [hunk 5]
```diff
@@ -58,7 +58,7 @@ func updateField(e *httpexpect.Expect, mID, fID, title, desc, key string, multip
...
```
Comment (yk-eukarya) [inline server/e2e/gql_field_test.go]:
```suggestion
			"modelId":      mID,
```

**Review by yk-eukarya — APPROVED** [review]
LGTM!

**Pull request merged**